samples/Samples.MultipleSchemas/Program.cs

```diff
@@ -16,21 +16,21 @@
 app.UseGraphQL<MultipleSchema.Cats.CatsSchema>("/cats/graphql");
 // configure the graphql endpoint at "/dogs/graphql"
 app.UseGraphQL<MultipleSchema.Dogs.DogsSchema>("/dogs/graphql");
-// configure Playground at "/cats"
+// configure Playground at "/cats/ui/playground" with relative link to api
 app.UseGraphQLPlayground(
     new GraphQL.Server.Ui.Playground.PlaygroundOptions
     {
-        GraphQLEndPoint = "/cats/graphql",
-        SubscriptionsEndPoint = "/cats/graphql",
+        GraphQLEndPoint = "../graphql",
```

Comment:
I don't know JS and this seems strange to me. I expecred that `../graphql` for `/cats/ui/playground` is `/cats/ui/graphql` like things for file paths.

Reply:
No, “graphql” would be the sibling file while “../graphql” is GraphQL within the parent folder. Note that no JS parsing is done to the GraphQL endpoint; only the subscription endpoint. And as both work, it shows that the code matches the browser interpretation of “../graphql”

Reply:
Does that make sense?

Reply:
I don't get it yet. What is a "parent folder" for `/cats/ui/playground`?

Reply:
| URL | Path | File | `img.jpg` resolves to | `../img.jpg` resolves to |
|-|-|-|-|-|
| `/` | `/` | (none) | `/img.jpg` | n/a |
| `/index.html` | `/` | `index.html` | `/img.jpg` | n/a |
| `/ui/` | `/ui/` | (none) | `/ui/img.jpg` | `/img.jpg` |
| `/ui/index.html` | `/ui/` | `index.html` | `/ui/img.jpg` | `/img.jpg` |
| `/ui/playground` | `/ui/` | `playground` | `/ui/img.jpg` | `/img.jpg` |

You can similarly apply `graphql` in place of `img.jpg`:

| URL | Path | File | `graphql` resolves to | `../graphql` resolves to |
|-|-|-|-|-|
| `/` | `/` | (none) | `/graphql` | n/a |
| `/index.html` | `/` | `index.html` | `/graphql` | n/a |
| `/ui/` | `/ui/` | (none) | `/ui/graphql` | `/graphql` |
| `/ui/index.html` | `/ui/` | `index.html` | `/ui/graphql` | `/graphql` |
| `/ui/playground` | `/ui/` | `playground` | `/ui/graphql` | `/graphql` |

Reply:
The above samples are just typical samples of how URLs work within any HTML.  There is nothing special about GraphQL or Playground that would affect how URLs are interpreted.  The only reason we need special code at all is because fully-qualified URLs are required for WebSocket communication.  So while we can pass relative URLs directly to `fetch`, we cannot use relative URLs for WebSocket connection requests.  So now for `fetch` (for POST requests), we just pass the unmodified relative (or absolute) URL while for WebSockets, we convert relative URLs to fully-qualified URLs.

Also, by testing `../graphql` for the `/cats/ui/playground` endpoint, we can be sure that the code works properly.  If it navigated up too many parents, `/graphql` would be attempted and would fail.  Likewise if it did not navigate up enough parents, `/cats/ui/graphql` would be attempted and would fail.

Reply:
I believe the reason fully-qualified URLs are required for WebSocket communication is because the protocol is "ws" or "wss" rather than "http" or "https".  Since the relative path of `../graphql` for `https://localhost/cats/ui/playground` resolves to `https://localhost/cats/graphql`, connection fails because it has the protocol of "https" rather than "wss".

Reply:
Ahh, I think I get it, I was looking at `/cats/ui/playground` as three-folder path cats->ui->playground so one step up would be cats->ui but playground is not a folder, it is a resource (file). Thanks.

Regarding that comment:
>  The only reason we need special code at all is because fully-qualified URLs are required for WebSocket communication. So while we can pass relative URLs directly to fetch, we cannot use relative URLs for WebSocket connection requests. So now for fetch (for POST requests), we just pass the unmodified relative (or absolute) URL while for WebSockets, we convert relative URLs to fully-qualified URLs.

I beleive it's worth to put it somewhere into the code.